pkg/server/bootstrap/identity.go

```diff
@@ -157,26 +160,48 @@ func wildcardIdentitiesResolver(ids *identities,
 				continue
 			}
 
-			apiExport, err := kcpClient.ApisV1alpha1().APIExports().Get(ctx, name, metav1.GetOptions{})
+			apiExportIdentity, err := apiExportIdentityProviderFn(ctx, name)
 			if err != nil {
 				errs = append(errs, err)
 				continue
 			}
-			if apiExport.Status.IdentityHash == "" {
+			if apiExportIdentity == "" {
 				errs = append(errs, fmt.Errorf("APIExport %s is not ready", name))
 				continue
 			}
 
 			ids.lock.Lock()
-			ids.groupResourceIdentities[gr] = apiExport.Status.IdentityHash
+			ids.groupResourceIdentities[gr] = apiExportIdentity
 			ids.lock.Unlock()
 
-			klog.V(2).Infof("APIExport %s|%s for resource %s.%s has identity %s", logicalcluster.From(apiExport), name, gr.Resource, gr.Group, apiExport.Status.IdentityHash)
+			klog.V(2).Infof("APIExport %s|%s for resource %s.%s has identity %s", tenancyv1alpha1.RootCluster, name, gr.Resource, gr.Group, apiExportIdentity)
 		}
 		return errorsutil.NewAggregate(errs)
 	}
 }
 
+func apiExportIdentityProviderFn(kcpClient kcpclient.Interface, kubeClusterClient kubernetes.ClusterInterface) func(ctx context.Context, apiExportName string) (string, error) {
+	return func(ctx context.Context, apiExportName string) (string, error) {
+		if kubeClusterClient != nil {
+			actualApiExportIdentitiesConfigMap, err := kubeClusterClient.Cluster(configshard.SystemShardCluster).CoreV1().ConfigMaps("default").Get(ctx, configMapName, metav1.GetOptions{})
+			if err == nil {
+				apiExportIdentity, found := actualApiExportIdentitiesConfigMap.Data[apiExportName]
+				if !found {
+					return "", fmt.Errorf("an identity entry for APIExport %s wasn't found in %s|%s configmap", apiExportName, configshard.SystemShardCluster, configMapName)
+				}
+				return apiExportIdentity, nil
+			}
+			// in case of any error just fallback and try to get an export directly
```

Comment:
L190 returns an error when the export is not found in the CM - in what sense is this fallback?

Reply:
we fall back when the cm wasn't found or we weren't able to get one. 

we could also fallback when an entry wasn't found but I'd let the controller change to update the cm - the resolver will retry.

Reply:
this would couple bootstrapping with the controller startup. Am not convinced that is good. Unnecessary complexity. Just fall through on incomplete CM and read from export.

Note: this is a rare case.

Reply:
no strong opinion here, I can update if we like it.

Reply:
> this would couple bootstrapping with the controller startup

it wouldn't as we create a config map with data

Reply:
> Unnecessary complexity

not falling back proves the controller actually works and it might be easier to debug as the source is easier to trace.